products/jbrowse-desktop/src/rootModel.ts

```diff
@@ -208,12 +376,21 @@ export default function rootModelFactory(pluginManager: PluginManager) {
       afterCreate() {
         addDisposer(
           self,
-          autorun(() => {
+          autorun(async () => {
             self.jbrowse.internetAccounts.forEach(account => {
               this.initializeInternetAccount(account.internetAccountId)
             })
           }),
         )
+        addDisposer(
+          self,
+          autorun(async () => {
+            if (self.indexingQueue.length > 0) {
+              await this.runIndexingJob()
+              this.dequeueIndexingJob()
+            }
+          }),
```

Comment:
could you add some error reporting so that the user knows if one of these failed? Probably a snackbar error dialog that includes a 'retry' button to requeue the job, and a button to open the config editor to edit the text indexing config. Also they need some way to re-run the job after editing the config.

Reply:
Should we add buttons to retry a failed indexing job from a widget instead? The snackbar material ui messages seem to be brief messages or notifications. We could also avoid having to send all the information needed to for indexing/queueing  to the snackbar messages system that we have in place.  In the current state of the PR, the user can go ahead and modify a track and then run the re-index track from the track menu items.

Reply:
Snackbars in our app don't have the ability to have buttons right now, but I created this issue to track that since it could be a good addition later: https://github.com/GMOD/jbrowse-components/issues/2866